mmeds/documents.py

```diff
@@ -35,43 +36,69 @@ class MMEDSDoc(men.Document):
     analysis_status = men.StringField(max_length=45)
     restart_stage = men.IntField()
     pid = men.IntField()
+    exit_code = men.IntField()
     files = men.DictField()
     config = men.DictField()
 
     # When the document is updated record the
     # location of all files in a new file
-    def save(self):
-        with open(str(Path(self.path) / 'file_index.tsv'), 'w') as f:
-            f.write('{}\t{}\t{}\n'.format(self.owner, self.email, self.access_code))
-            f.write('Key\tPath\n')
-            for key, file_path in self.files.items():
-                # Skip non existent files
-                if file_path is None:
-                    continue
-                # If it's a key for an analysis point to the file index for that analysis
-                elif isinstance(file_path, dict):
-                    f.write('{}\t{}\n'.format(key, Path(self.path) / key / 'file_index.tsv'))
-                # Otherwise just write the value
-                else:
-                    f.write('{}\t{}\n'.format(key, file_path))
-        with open(DOCUMENT_LOG, 'a') as f:
-            f.write('-\t'.join([str(type(self)), self.owner, 'Upload', 'Finished', self.path, self.access_code]) + '\n')
-        super().save()
+    def save(self, **kwargs):
+        super().save(**kwargs)
+        if self.path is not None:
+            with open(str(Path(self.path) / 'file_index.tsv'), 'w') as f:
```

Comment:
I'm still not a big fan of using ```f``` when dealing with a specific file.

Reply:
Okay well PEP8 disagrees with you. I don't see how it's unclear what file it's referring to. If I had multiple files open at the same time I'd agree with you, but in this case `f` is known to refer to the currently open file

Reply:
It's not that it's ambiguous, but I feel like giving the file name when it is a specific file improves the readability of the code. But hey, that's my take and given that PEP8 disagrees with me, I'm not obliging you to change it.